code/modules/mob/living/simple_animal/hostile/goose.dm

```diff
@@ -58,24 +59,44 @@
 	QDEL_NULL(goosevomit)
 	return ..()
 
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/examine(user)
+	..()
+	to_chat(user, "<span class='notice'>Somehow, it still looks hungry.</span>")
+
 /mob/living/simple_animal/hostile/retaliate/goose/vomit/attacked_by(obj/item/O, mob/user)
 	feed(O)
 
 /mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/feed(obj/item/O)
 	if(istype(O, /obj/item/reagent_containers/food))
-		O.forceMove(src)
-		playsound(src,'sound/items/eatfood.ogg', rand(10,50), 1)
-		vomitCoefficient ++
+		var/obj/item/reagent_containers/food/tasty = O
+		if (tasty.foodtype & GROSS)
+			visible_message("<span class='notice'>[src] hungrily gobbles up \the [tasty]!</span>")
+			tasty.forceMove(src)
+			playsound(src,'sound/items/eatfood.ogg', 70, 1)
+			vomitCoefficient ++
+			vomitTimeBonus += 5
+		else
+			visible_message("<span class='notice'>[src] refuses to eat \the [tasty].</span>")
 
 /mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit()
 	var/turf/T = get_turf(src)
 	playsound(get_turf(src), 'sound/effects/splat.ogg', 50, 1) //yes getting the turf twice is necessary fuck off
-	T.add_vomit_floor(src)
-	for (var/obj/item/consumed in contents)
-		if(istype(consumed, /obj/item/reagent_containers/food))
-			consumed.forceMove(T)
-			var/destination = get_edge_target_turf(T, pick(GLOB.alldirs)) //Pick a random direction to toss them in
-			consumed.throw_at(destination, 2, 3) //Thow the food at a random tile 3 spots away
+	if (prob(50))
+		var/obj/item/reagent_containers/food/consumed = locate() in contents //Barf out a single food item from our guts
+		if (consumed)
+			barf_food(consumed)
+	else
+		T.add_vomit_floor(src)
+	
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/barf_food(var/atom/A)
+	if(istype(A, /obj/item/reagent_containers/food))
+		var/turf/T = get_turf(src)
+		var/obj/item/reagent_containers/food/consumed = A
```

Comment:
`consumed.foodtype |= GROSS`
and
`consumed.description += " It looks covered in bile"`

because the fucker literaly puked it out

Reply:
Buffing gross food eaters so they can turn anything into gross food smh